src/withPlugins.ts

```diff
@@ -1,9 +1,20 @@
 import { Plugins, DiceRule } from './rules/types';
 import createDiceRoller, { defaultPlugins } from './createDiceRoller';
+import defaultRandom, { Random } from "./util/random"
 
-function withPlugins(...plugins: DiceRule<any>[]) {
+function withPlugins(...plugins: DiceRule<any>[]): ReturnType<typeof createDiceRoller>
+function withPlugins(plugins: DiceRule<any>[], random?: Random): ReturnType<typeof createDiceRoller>
+function withPlugins(...args: any[]): ReturnType<typeof createDiceRoller> {
+  let plugins: DiceRule<any>[];
+  let random = defaultRandom
+  if (Array.isArray(args[0])) {
+    plugins = args[0]
+    random = args[1] || defaultRandom
+  } else {
+    plugins = args
```

Comment:
I should definitely add `Prettier` to CI, there are some semicolon inconsistencies in here. I wouldn't worry about this now, just writing my thoughts down haha.

Reply:
I will send a pr with some improvements and pre-commit hooks that auto run prettier on changed files :)

Reply:
Thanks!